kroxylicious/src/test/java/io/kroxylicious/proxy/internal/FilterHandlerTest.java

```diff
@@ -607,6 +612,80 @@ void testDropResponse() {
      */
     @Test
     void sendRequest() {
+        var oobRequestBody = new FetchRequestData();
+        var snoopedOobRequestResponseStage = new AtomicReference<CompletionStage<ResponseHeaderAndApiMessage<FetchRequestData>>>();
+        ApiVersionsRequestFilter filter = (apiVersion, header, request, context) -> {
+            assertNull(snoopedOobRequestResponseStage.get(), "Expected to only be called once");
+            snoopedOobRequestResponseStage.set(context.sendRequest(new RequestHeaderData(), oobRequestBody));
+            return snoopedOobRequestResponseStage.get()
+                    .thenCompose(u -> context.forwardRequest(header, request));
+        };
+
+        buildChannel(filter);
+
+        // trigger filter
+        var requestFrame = writeRequest(new ApiVersionsRequestData());
+
+        // verify filter has sent the send request.
+        InternalRequestFrame<?> propagatedOobRequest = channel.readOutbound();
+        assertThat(propagatedOobRequest.body()).isEqualTo(oobRequestBody);
+        assertThat(propagatedOobRequest.header()).isNotNull();
+
+        // verify oob request response future is in the expected state
+        assertThat(snoopedOobRequestResponseStage).isNotNull();
+        var snoopedOobRequestResponseFuture = toCompletableFuture(snoopedOobRequestResponseStage.get());
+        assertThat(snoopedOobRequestResponseFuture).withFailMessage("out-of-band request response future in wrong state").isNotDone();
```

Comment:
```suggestion
        assertThat(snoopedOobRequestResponseFuture).withFailMessage("out-of-band request response future was expected to be pending but it is done.").isNotDone();
```

Its clear when reading the code but if I was reading a failure message I don't think it would be clear why the test had failed.

Reply:
I thought this suggestion was accepted but the diff seems to show it being reverted?

Reply:
I made a mistake with Git on Friday.  I must have reverted the change my mistake.